whole-tale/ngx-dashboard
# Fix notification stream reconnect spam

## Problem
Girder's event stream API does not send periodic keep-alive message over the stream. This causes the client to think that the stream is inactive and disconnect every 45000 milliseconds with a thrown error. This error is innocuous, however, since the client immediately reconnects and operations resume as expected.

Note that this issue was also prevalent in the Ember implementation of the dashboard. From the client side, even forcing a disconnect and reconnecting again manually every 30 seconds does not appear to successfully workaround the error message spam. 

Fixes #5 

## Approach
Just watch for the exact error message(s) that are spammed and squelch them. I do not believe that these particular errors would ever be indicative of a real problem, and other notifications appear to still work as intended under these conditions.

Thorough testing may be required to ensure that even more notifications are not being mysteriously skipped or dropped, as has been previously reported. I have been unable to reliably reproduce this behavior.

## How to Test
Prerequisites: at least one Tale created

1. Checkout this branch locally, rebuild the dashboard
2. Login to the WholeTale Dashboard
3. Open your browser's Developer Console to the `Console` tab and wait for a minute or two
    * You should **NOT** see "No activity for 45000 milliseconds. Reconnect" appearing every 45 seconds in the console
4. Navigate to the Tale Catalog, hover over a Tale that you own, and click View
    * You should be brought to the Run view
5. At the top-right, click "Run Tale"
    * You should see your Tale begin to start
    * After a moment, you should see the notification stream appear at the top-right
6. Check the developer console again 
    * You still should **NOT** see "No activity for 45000 milliseconds. Reconnect" appearing every 45 seconds in the console
